### PR TITLE
Remove CORS support

### PR DESCRIPTION
**Pull request by cnunciato**
This change removes CORS support from builder-api and removes the configurability of the `habitat_api_url` setting. It also adds support for proxying Builder API services in development in order to work around this new restriction.

Signed-off-by: Christian Nunciato <chris@nunciato.org>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-web/package-lock.json [hunk 1]
```diff
@@ -4302,6 +4307,256 @@
         "extend": "3.0.1"
       }
     },
+    "http-proxy-middleware": {
```
Comment (chefsalim) [inline components/builder-web/package-lock.json]: Are we including this in the prod package as well?  Not clear whether this is dev only..

File: components/builder-web/bs-config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const proxyMiddleware = require('http-proxy-middleware');
```
Comment (chefsalim) [inline components/builder-web/bs-config.js]: Are there any specific steps for devs to take advantage of this?

File: components/builder-web/package-lock.json (hunk 1 above)
Comment (cnunciato) [reply components/builder-web/package-lock.json]: Nope, it's a `devDependency`, so it's only needed in dev.

File: components/builder-web/bs-config.js (hunk 2 above)
Comment (cnunciato) [reply components/builder-web/bs-config.js]: This makes me wonder whether the API URL even needs to be configurable anymore. I don't think it does, so I'll update this PR to remove that as well.

**Comment by cnunciato** [comment]
All right @chefsalim, this should take care of what's needed for `builder-api-proxy`. I'm looking at removing the CORS support in `builder-api` now.

**Comment by elliott-davis** [comment]
We need to add cors back into the API otherwise GitHub auth is going to break _i think_

**Comment by chefsalim** [comment]
Both webhooks and auth are ok on acceptance

**Pull request merged**